src/main/MQ2ImGuiConsole.cpp

```diff
@@ -27,1504 +27,1638 @@
-			{
-				ImGuiID topId;
-				ImGui::DockBuilderSplitNode(dockspaceNode->ID, ImGuiDir_Up, 0.30f, &topId, nullptr);
-				topNode = ImGui::DockBuilderGetNode(topId);
-			}
+				if (consoleWindow)
+				{
+					ImGui::SetWindowDock(consoleWindow, topNode->ID, ImGuiCond_Always);
+					consoleWindow->DockOrder = -1;
+				}
+				else
+				{
+					ImGui::SetNextWindowDockID(topNode->ID, ImGuiCond_FirstUseEver);
+				}
 
-			if (consoleWindow)
-			{
-				ImGui::SetWindowDock(consoleWindow, topNode->ID, ImGuiCond_Always);
-				consoleWindow->DockOrder = -1;
+				ImGui::DockBuilderFinish(s_dockspaceId);
 			}
-			else
+			s_resetConsolePosition = false;
+
+			gImGuiConsole->Draw(&s_consoleVisible);
+
+			if (shouldSetFocusToMainViewport)
 			{
-				ImGui::SetNextWindowDockID(topNode->ID, ImGuiCond_FirstUseEver);
-			}
+				// activate main viewport in case it isn't currently in focus
+				ImGuiViewport* mainViewport = ImGui::GetMainViewport();
 
-			ImGui::DockBuilderFinish(s_dockspaceId);
+				if (ImGui::GetCurrentWindowRead()->Viewport->ID != mainViewport->ID)
+				{
+					// Activate the main viewport window.
+					::SetActiveWindow((HWND)mainViewport->PlatformHandle);
+				}
+			}
 		}
-		s_resetConsolePosition = false;
+	}
 
-		gImGuiConsole->Draw(&s_consoleVisible);
+	void MQConsoleCommand(SPAWNINFO* pChar, char* Line)
+	{
+		char szCommand[MAX_STRING] = { 0 };
+		GetArg(szCommand, Line, 1);
 
-		if (shouldSetFocusToMainViewport)
+		if (!_stricmp("clear", szCommand))
 		{
-			// activate main viewport in case it isn't currently in focus
-			ImGuiViewport* mainViewport = ImGui::GetMainViewport();
+			if (gImGuiConsole != nullptr)
+				gImGuiConsole->ClearLog();
 
-			if (ImGui::GetCurrentWindowRead()->Viewport->ID != mainViewport->ID)
-			{
-				// Activate the main viewport window.
-				::SetActiveWindow((HWND)mainViewport->PlatformHandle);
-			}
+			return;
 		}
-	}
-}
 
-void MQConsoleCommand(SPAWNINFO* pChar, char* Line)
-{
-	char szCommand[MAX_STRING] = { 0 };
-	GetArg(szCommand, Line, 1);
+		if (!_stricmp("toggle", szCommand))
+		{
+			gbToggleConsoleRequested = true;
+			return;
+		}
 
-	if (!_stricmp("clear", szCommand))
-	{
-		if (gImGuiConsole != nullptr)
-			gImGuiConsole->ClearLog();
+		if (!_stricmp("show", szCommand))
+		{
+			gbSetConsoleVisibilityRequest = true;
+			return;
+		}
 
-		return;
-	}
+		if (!_stricmp("hide", szCommand))
+		{
+			gbSetConsoleVisibilityRequest = false;
+			return;
+		}
 
-	if (!_stricmp("toggle", szCommand))
-	{
-		gbToggleConsoleRequested = true;
-		return;
+		WriteChatf("Usage: /mqconsole [command]");
+		WriteChatf("  Commands: clear, toggle, show, hide");
 	}
 
-	if (!_stricmp("show", szCommand))
+	int FontNumberFromString(std::string fontString)
 	{
-		gbSetConsoleVisibilityRequest = true;
-		return;
+		ImGuiIO& io = ImGui::GetIO();
+		for (int i = 0; i < io.Fonts->Fonts.Size; i++)
+		{
+			std::string fontstr = io.Fonts->Fonts[i]->GetDebugName();
+			if (strcmp(fontstr.c_str(), fontString.c_str()) == 0)
```

Comment:
you took a const char*, turned it into a std::string, and then took the const char* back out.

This function can take a string_view and use ci_equals from the mq string library.

function name isn't correct. "FontNumberFromString" - what about the font number? what are we doing with it? Getting it? converting it? etc